From dfa8e50941c36a8166d1453a94674ad4d76e4e76 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Fri, 23 May 2014 13:54:35 +0800
Subject: [PATCH 30/30] ENGR00304168 fix show black screen for magnification
 gesture when play video.

the root cause is computing source crop reduce transparent region
but not for visible region screen when using HWComposer to do composite.
use visibleNonTransparentRegion instead of visibleRegion to set
visible region screen to HWComposer.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 services/surfaceflinger/Layer.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index 909a509..2329c69 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -414,7 +414,8 @@ void Layer::setPerFrameData(const sp<const DisplayDevice>& hw,
     // Apply this display's projection's viewport to the visible region
     // before giving it to the HWC HAL.
     const Transform& tr = hw->getTransform();
-    Region visible = tr.transform(visibleRegion.intersect(hw->getViewport()));
+    Region visible = tr.transform(visibleNonTransparentRegion.intersect(
+                                   hw->getViewport()));
     layer.setVisibleRegionScreen(visible);
 
     // NOTE: buffer can be NULL if the client never drew into this
-- 
1.8.0

